### PR TITLE
Settings: Metric Configuration (1/n)

### LINKED ISSUE #63: [Frontend] New Settings Page: Metric Configuration (1/n)

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Removes the `/metrics` route (and from the Menu) and refactors the MetricsView component to match the updated design (up until the detailed metrics view which is currently styled the same as before]). This will help set up the next phase of work within the Metric Configuration which will be updating the detailed metrics view per the new design.

Demo of current change:

https://user-images.githubusercontent.com/59492998/194413204-9c17a55d-be9f-40a4-869c-8fd44cef3bb6.mov

## Related issues

Closes #63 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by mxosman** [comment]
Splitting this up in this way so it's not a huge diff.

File: publisher/src/components/MetricsView/MetricsView.tsx [hunk 1]
```diff
@@ -827,129 +797,78 @@ export const MetricsView: React.FC = observer(() => {
...
+                activeMetricKey={activeMetricKey}
+                setActiveMetricKey={setActiveMetricKey}
+              />
+            ))}
+
+          {/* Metric Configuration */}
+          {activeMetricKey && (
+            <MetricConfigurationDisplay>
+              <BackToMetrics onClick={() => setActiveMetricKey("")}>
+                ← Back to Metrics
```
Comment (terryttsai) [inline publisher/src/components/MetricsView/MetricsView.tsx:851]: This back navigation arrow sort of implies to me that you can press the browser back button to navigate back to metrics, like for example in the data entry page, but it doesn't do that. 

It's probably fine, I can't think of a quick way to make this more consistent anyways.

**Review by terryttsai — COMMENTED** [review]
Are we removing the ability for users to see data visualizations? Or are we adding it back in a later PR?

**Comment by mxosman** [comment]
> Are we removing the ability for users to see data visualizations? Or are we adding it back in a later PR?

Great question! I'm not sure. We should ask Humphrey & Juan. In the mocks, there's a link to a `Data` page in the menu, which is where I assume it will live? Not 100% sure.

File: publisher/src/components/MetricsView/MetricsView.tsx (hunk 1 above)
Comment (mxosman) [reply publisher/src/components/MetricsView/MetricsView.tsx:851]: Oh interesting! I don't see it that way - if I'm coming from the list of metrics, it looks contained within & relevant to the metric config display that I interpret it as back to the list of metrics completely unrelated to the browser back button. Feel free to bring this up on the Figma design!

Comment (mxosman) [reply publisher/src/components/MetricsView/MetricsView.tsx:851]: Maybe saying something like "Back to List of Metrics" would be helpful? Not sure.

The mocks say "Back to All Metrics" but we made a decision a while back on that original sprint to remove the `All Metrics` filter as it didn't feel necessary. Not sure.

**Comment by terryttsai** [comment]
Guess it's not a blocker since you're working for this on a feature branch right? And we won't be releasing that feature branch to production until it's ready?

**Review by terryttsai — APPROVED** [review]

File: publisher/src/components/MetricsView/MetricsView.tsx (hunk 1 above)
Comment (terryttsai) [reply publisher/src/components/MetricsView/MetricsView.tsx:851]: Good point, I'll comment on the figma! Meanwhile I don't think this should be a blocker and the current design is good with me!

**Comment by mxosman** [comment]
> Guess it's not a blocker since you're working for this on a feature branch right? And we won't be releasing that feature branch to production until it's ready?

Yeah - exactly! And, if it does end up going in its own page - I would imagine we'd get a mock up of that page too (or we should request one).

I'm also still trying to figure out the best way to split these PRs up - open to any suggestions - frontend is always hard for me to split because of how intertwined everything is.

**Comment by lilidworkin** [comment]
@mxosman I actually LOVE how you split this up! Thank you. So much easier to reason about and review!

re: data page -- great catch @terryttsai ! Can't believe I didn't think of this during design review! Can you or @mxosman post in @justice-counts and tag Humphrey/Juan about this? If we can get a design for how to move the existing data viz into a new page, that might be a perfect project for a frontend contractor.

> Guess it's not a blocker since you're working for this on a feature branch right? And we won't be releasing that feature branch to production until it's ready?

Yes @terryttsai  exactly -- I think this is a really nice feature of the feature branch system -- it should hopefully make it a lot easier to merge in incomplete PRs like this, which we wouldn't want to get into production yet, but we've vetted all the logic as a solid building block.

**Comment by lilidworkin** [comment]
@mxosman just checking, do the uploaded files now show up on their own in the uploaded files tab?

**Comment by terryttsai** [comment]
Loving the feature branches system already!

**Pull request merged**